res/controllers/Reloop-Digital-Jockey-2-IE.scripts.js

```diff
@@ -0,0 +1,876 @@
+        } else {
+            engine.setValue(this.group, "back", this.isPress(channel, control, value, status));
+        }
+    },
+    unshift: function() {
+        this.inSetValue(false);     // bug? inKey was not reset by default on unshift()
+        this.inKey = "rate_temp_down";
+        this.disconnect();
+        this.outKey = "rate_temp_down";
+        this.connect();
+        this.trigger();
+    },
+    /* The shifted action is disabled here as it has been remapped
+       to '[Library]GoToItem' normal binding in the xml file.
+
+    shift: function() {
+        this.inKey = "rate_temp_down_small";
+        this.disconnect();
+        this.outKey = "rate_temp_down_small";
+        this.connect();
+        this.trigger();
+    },*/
+});
+
+RDJ2.BendPlusButton = function(options) {
+    components.Button.call(this, options);
+};
+RDJ2.BendPlusButton.prototype = new components.Button({
+    input: function(channel, control, value, status) {
+        var isPlaying = engine.getValue(this.group, "play");
+        if (isPlaying) {
+            engine.setValue(this.group, "fwd", false);
+            this.inSetValue(this.isPress(channel, control, value, status));
+        } else {
+            engine.setValue(this.group, "fwd", this.isPress(channel, control, value, status));
+        }
+    },
+    unshift: function() {
+        this.inSetValue(false);     // bug? inKey was not reset by default on unshift()
+        this.inKey = "rate_temp_up";
+        this.disconnect();
+        this.outKey = "rate_temp_up";
+        this.connect();
+        this.trigger();
+    },
+    shift: function() {
+        this.inKey = "rate_temp_up_small";
+        this.disconnect();
+        this.outKey = "rate_temp_up_small";
+        this.connect();
+        this.trigger();
+    },
+});
+
+/* Jog Mode */
+
+RDJ2.JogModeSelector = function(number, searchMidiCtrl, scratchMidiCtrl, fxDryWetMidiCtrl) {
+    this.number = number;
+    this.searchMidiCtrl = searchMidiCtrl;
+    this.scratchMidiCtrl = scratchMidiCtrl;
+    this.fxDryWetMidiCtrl = fxDryWetMidiCtrl;
+    this.jogMode = RDJ2.JOGMODES.normal;
+    this.lastNonTraxJogMode = this.jogMode;
+    this.input = this.inputNormal;
+
+    components.Component.call(this);
+    this.updateControls();
+};
+RDJ2.JogModeSelector.prototype = new components.Component({
+    updateControls: function() {
+        var searchValue = this.jogMode === RDJ2.JOGMODES.search ? RDJ2.MIDI_ON : RDJ2.MIDI_OFF;
+        var scratchValue = this.jogMode === RDJ2.JOGMODES.vinyl ? RDJ2.MIDI_ON : RDJ2.MIDI_OFF;
+        var fxDryWetValue = this.jogMode === RDJ2.JOGMODES.fxdrywet ? RDJ2.MIDI_ON : RDJ2.MIDI_OFF;
+
+        if (midi.sendShortMsg) {
+            midi.sendShortMsg(0x90, this.searchMidiCtrl, searchValue);
+            midi.sendShortMsg(0x90, this.scratchMidiCtrl, scratchValue);
+            midi.sendShortMsg(0x90, this.fxDryWetMidiCtrl, fxDryWetValue);
+        }
```

Comment:
this check should be unnecessary. `midi.sendShortMsg` should always be available.

Reply:
I thought so. And this is how it was originally before running ctest. However one test was failing:
```
The following tests FAILED:
         93 - ControllerPresetValidationTest.MidiPresetsValid (Failed)
```
and I looked into the test code and the js code and found out that somehow this method was not available because some specific test initialization. I had no idea how to fix it other way than by introducing this `if (midi.sendShortMsg)` check. I know this is probably not the best solution...

Reply:
Thats strange. I'll look into it.

Reply:
Great, this was the commit fixing the issue:
https://github.com/mixxxdj/mixxx/pull/4614/commits/e7e88044b2f88a80fbba091ae1e4f11151d4e557